esrally/track/loader.py

```diff
@@ -77,19 +83,17 @@ def register_track_processor(self, processor):
             processor.decompressor = Decompressor()
         self.track_processors.append(processor)
 
+    def processors(self):
+        if not self.track_processors:
```

Comment:
it's a bit confusing but the `TaskFilterTrackProcessor` and `TestModeTrackProcessor` are executed separately from this workflow.  Otherwise somewhere around this code block we need to say "always do these two but sometimes do this one." The code got kind of sloppy so I still invoke those two in [line 200/204] in a separate, dedicated TPR instance